ss_admin_console/main/default/classes/SSACDataFilter.cls

```diff
@@ -0,0 +1,115 @@
+public with sharing class SSACDataFilter {
+    private SObjectType sObjectApiName;
+    private List<SObject> records;
+    private List<FieldReference> keyFields;
+    private List<FieldReference> dataFields;
+
+    public SSACDataFilter(SObjectType sObjectApiName, List<FieldReference> keyFields, List<FieldReference> dataFields) {
+        this.sObjectApiName = sObjectApiName;
+        this.keyFields = keyFields;
+        this.dataFields = dataFields; 
+    }
+
+    // filter method for determining new, dublicated and existing records
+    public List<SSACResult> filter(List<SObject> records) {
+        List<SSACResult> results = new List<SSACResult>();
+        List<SObject> existentRecords = getExistentRecords(records);
+        Map<String, SObject> existentMap = generateExistentMap(existentRecords);
+       
+        Set<FieldReference> nonKeyFields = new Set<FieldReference>(dataFields.clone());
+        nonKeyFields.removeAll(this.keyFields);
+        
+        for (SObject record : records) {
+            Integer matchedNonKeyFields = 0;
+            String recordKeysFingerprint = '';
+            for (FieldReference keyField : keyFields) {
+                recordKeysFingerprint += String.valueOf(keyField.getFrom(record)).toLowerCase() + '_';
+            }
+
+            recordKeysFingerprint = recordKeysFingerprint.removeEnd(recordKeysFingerprint.right(1));
+            if (existentMap.get(recordKeysFingerprint) != null) {
```

Comment:
Extract `existentMap.get(recordKeysFingerprint)` to local variable and reuse it later (e.g. on line 32)